### PR TITLE
Updates GuildModifyRequest

### PR DESCRIPTION
**Pull request by Shadorc**
Also changes all fields to `Possible<Optional>` because the doc says `All parameters to this endpoint are optional`

**Justification:** https://github.com/discord/discord-api-docs/pull/1764

### CODE REVIEW

File: src/main/java/discord4j/discordjson/json/GuildModifyRequest.java [hunk 1]
```diff
@@ -17,34 +18,53 @@
         return ImmutableGuildModifyRequest.builder();
     }
 
-    Possible<String> name();
+    Possible<Optional<String>> name();
```
Comment (darichey) [inline src/main/java/discord4j/discordjson/json/GuildModifyRequest.java:21]: I don't think `name` is nullable?

File: src/main/java/discord4j/discordjson/json/GuildModifyRequest.java [hunk 2]
```diff
@@ -17,34 +18,53 @@
...
     @JsonProperty("explicit_content_filter")
-    Possible<Integer> explicitContentFilter();
+    Possible<Optional<Integer>> explicitContentFilter();
 
     @JsonProperty("afk_channel_id")
     Possible<Optional<String>> afkChannelId();
 
     @JsonProperty("afk_timeout")
-    Possible<Integer> afkTimeout();
+    Possible<Optional<Integer>> afkTimeout();
```
Comment (darichey) [inline src/main/java/discord4j/discordjson/json/GuildModifyRequest.java:38]: Also not nullable

File: src/main/java/discord4j/discordjson/json/GuildModifyRequest.java [hunk 3]
```diff
@@ -17,34 +18,53 @@
...
 
     Possible<Optional<String>> icon();
 
     @JsonProperty("owner_id")
-    Possible<String> ownerId();
+    Possible<Optional<String>> ownerId();
```
Comment (darichey) [inline src/main/java/discord4j/discordjson/json/GuildModifyRequest.java:43]: Same

File: src/main/java/discord4j/discordjson/json/GuildModifyRequest.java [hunk 4]
```diff
@@ -17,34 +18,53 @@
...
+    Possible<Optional<String>> discoverySplash();
+
     Possible<Optional<String>> banner();
 
     @JsonProperty("system_channel_id")
-    Possible<String> systemChannelId();
+    Possible<Optional<String>> systemChannelId();
+
+    @JsonProperty("system_channel_flags")
+    Possible<Optional<Integer>> systemChannelFlags();
```
Comment (darichey) [inline src/main/java/discord4j/discordjson/json/GuildModifyRequest.java:56]: Same

File: src/main/java/discord4j/discordjson/json/GuildModifyRequest.java [hunk 5]
```diff
@@ -17,34 +18,53 @@
...
+    @JsonProperty("rules_channel_id")
+    Possible<Optional<String>> rulesChannelId();
+
+    @JsonProperty("public_updates_channel_id")
+    Possible<Optional<String>> publicUpdatesChannelId();
+
+    @JsonProperty("preferred_locale")
+    Possible<Optional<String>> preferredLocale();
+
+    Possible<Optional<List<String>>> features();
```
Comment (darichey) [inline src/main/java/discord4j/discordjson/json/GuildModifyRequest.java:67]: Same

**Review by darichey — CHANGES_REQUESTED** [review]
There's a few fields that I don't think are nullable based on the docs diff. Lmk if I missed something

**Review by Shadorc — COMMENTED** [review]
Fixed

**Review by darichey — APPROVED** [review]

**Pull request merged**